src/web/app/pages-session/session-submission-page/session-submission-page.component.spec.ts

```diff
@@ -30,17 +578,520 @@ describe('SessionSubmissionPageComponent', () => {
+          questionDescription: testMcqQuestionSubmissionForm2.questionDescription,
+          questionDetails: testMcqQuestionSubmissionForm2.questionDetails,
+          questionType: testMcqQuestionSubmissionForm2.questionType,
+          giverType: testMcqQuestionSubmissionForm2.giverType,
+          recipientType: testMcqQuestionSubmissionForm2.recipientType,
+          numberOfEntitiesToGiveFeedbackToSetting:
+            testMcqQuestionSubmissionForm2.numberOfEntitiesToGiveFeedbackToSetting,
+          customNumberOfEntitiesToGiveFeedbackTo: testMcqQuestionSubmissionForm2.customNumberOfEntitiesToGiveFeedbackTo,
+          showResponsesTo: testMcqQuestionSubmissionForm2.showResponsesTo,
+          showGiverNameTo: testMcqQuestionSubmissionForm2.showGiverNameTo,
+          showRecipientNameTo: testMcqQuestionSubmissionForm2.showRecipientNameTo,
+        },
+      ],
+    };
+    const testFeedbackQuestionRecipients: FeedbackQuestionRecipients = {
+      recipients: [
+        {
+          name: 'Barry Harris',
+          identifier: 'barry-harris-id',
+        },
+        {
+          name: 'Gene Harris',
+          identifier: 'gene-harris-id',
+        },
+      ],
+    };
+    const testExistingResponses: FeedbackResponsesResponse = {
+      responses: [testResponse4, testResponse5],
+    };
+
+    spyOn(feedbackQuestionsService, 'getFeedbackQuestions').and.returnValue(of(testFeedbackQuestions));
+    spyOn(feedbackQuestionsService, 'loadFeedbackQuestionRecipients')
+        .and.returnValue(of(testFeedbackQuestionRecipients));
+    spyOn(feedbackResponsesService, 'getFeedbackResponse').and.returnValue(of(testExistingResponses));
+
+    component.loadFeedbackQuestions();
+
+    expect(component.questionSubmissionForms.length).toEqual(1);
+    expect(component.questionSubmissionForms[0]).toEqual(testMcqQuestionSubmissionForm2);
+  });
+
+  it('should get comment model', () => {
+    const testComment: FeedbackResponseComment = {
+      commentGiver: 'comment giver',
+      lastEditorEmail: 'last-editor@email.com',
+      feedbackResponseCommentId: 5,
+      commentText: 'comment text',
+      createdAt: 10000000,
+      lastEditedAt: 10000000,
+      isVisibilityFollowingFeedbackQuestion: true,
+      showGiverNameTo: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS],
+      showCommentTo: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS],
+    };
+    const expectedCommentModel: CommentRowModel = {
+      originalComment: testComment,
+      commentEditFormModel: {
+        commentText: testComment.commentText,
+        isUsingCustomVisibilities: false,
+        showCommentTo: [],
+        showGiverNameTo: [],
+      },
+      timezone: '',
+      isEditing: false,
+    };
+    const commentModel: CommentRowModel = component.getCommentModel(testComment);
+    expect(commentModel).toEqual(expectedCommentModel);
+  });
+
+  it('should check that there are responses to submit', () => {
```

Comment:
Maybe you can just combine this testcase with the `'should load feedback questions and responses'` testcase